src/backend/distributed/commands/utility_hook.c

```diff
@@ -1627,17 +1664,93 @@ RunPreprocessMainDBCommand(Node *parsetree, const char *queryString)
 static void
 RunPostprocessMainDBCommand(Node *parsetree)
 {
+	if (IsStatementSupportedInNonMainDb(parsetree) &&
+		StatementRequiresMarkDistributedFromNonMainDb(parsetree))
+	{
+		MarkObjectDistributedInNonMainDb(parsetree);
+	}
+}
+
+
```

Comment:
Those four functions should be defined in the following order as we agreed on in https://github.com/citusdata/citus/pull/7347, due to the order we call them:

Their declarations (see top of this file) should also follow this ordering.

static .. IsStatementSupportedInNonMainDb
static .. StatementRequiresMarkDistributedFromNonMainDb
static .. MarkObjectDistributedInNonMainDb
static .. GetObjectInfo

Reply:
Fixed

Reply:
You're right. Fixed function definitions as well